### ISSUE TITLE
meson 0.46.0 - invalid .pc file written for libgxps

### ISSUE THREAD

**Issue by DimStar77** [mesonbuild/meson#3479]
This is a sample,other builds can excerpt the same.

libgxps.pc generated with meson < 0.46.0 (tested with 0.45.0) contains:

```
Requires: gobject-2.0 >= 2.36.0, gio-2.0 >= 2.36.0, libarchive >= 2.8.0, cairo >= 1.10.0
```

The same source build using meson 0.46.0 contains:

```
Requires: gobject-2.0 >= 2.36.0,gio-2.0>=2.36.0,libarchive>=2.8.0,cairo>;=1.10.0
```

which in turn is not understood by pkgconfig:

```
$ pkg-config --libs libgxps
Package gio-2.0>=2.36.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `gio-2.0>=2.36.0.pc'
to the PKG_CONFIG_PATH environment variable
Package 'gio-2.0>=2.36.0', required by 'libgxps', not found
```

**Comment by DimStar77** [comment]
libgxps' meson.build contains:


```
# Generate the pkg-config file
pkgg = import('pkgconfig')

pkgg.generate(libraries: gxps,
              version: gxps_version,
              name: 'libgxps',
              description: 'XPS Documents library',
              requires: 'gobject-2.0 >= ' + glib_req + ', gio-2.0 >= ' + glib_req + ', libarchive >= ' + archive_req + ', cairo >= ' + cairo_req)
```

**Comment by DimStar77** [comment]
in case it matters (and/or more samples are needed): libgit2-glib's .pc file shows the same (also uses ```pkg.generate()```)

**Comment by xclaesse** [comment]
As a side note, with meson 0.46 the generator will set version automatically with the one you passed to dependency().

**Comment by jpakkane** [comment]
Can you test if the linked PR fixes this for you?

**Comment by DimStar77** [comment]
backported cbc2f2d into our meson package, and the issues identified earlier are now resolved (tested with libgxps/evince and libgit2-glib

**Comment by rezso** [comment]
Another problem with .pc file generation.
If I build libxkbcommon 0.8.0 with meson 0.44, the xkbcommon.pc contains the expected Libs line:
``Libs: -L${libdir} -lxkbcommon``
But if I build ibxkbcommon 0.8.0 with meson 0.48, the pc file contains a bad Libs line:
``Libs: -L${libdir} -lxkbcommon-internal``
The libxkbcommon-internal is an static library, the final shared library is the libxkbcommon.so.

**Comment by xclaesse** [comment]
Can you please open a separate issue for that, and link to your meson.build? Also I'd that a regression, in 0.46.0 ?

**Comment by rezso** [comment]
I opened a new issue: https://github.com/mesonbuild/meson/issues/3509.
And yes, this is bad in meson 0.46, the 0.48 is a typo.

**Comment by royb3** [comment]
Actually, the documentation tells you to provide a list of strings. I had the same issue while trying to use libgit2-glib.

http://mesonbuild.com/Pkgconfig-module.html
